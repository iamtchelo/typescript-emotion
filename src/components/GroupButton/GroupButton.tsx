import * as React from 'react';
import { GroupButtonWrapper } from './group-button.elements';

export interface GroupButtonProps {
  direction?: 'left' | 'right';
}

const GroupButton: React.SFC<GroupButtonProps> = ({ direction = 'right', children }) => (
  <GroupButtonWrapper direction={direction}>
    {children}
  </GroupButtonWrapper>
);

export default GroupButton;
