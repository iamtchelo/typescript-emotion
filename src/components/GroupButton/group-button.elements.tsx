import styled from 'react-emotion';
import { GroupButtonProps } from './GroupButton';

export const GroupButtonWrapper = styled('div')`
  display: flex;
  padding: 40px 0;
  flex-direction: ${({ direction }: GroupButtonProps) => {
    return direction === 'left' ? 'row' : 'row-reverse'}
  };
`;
