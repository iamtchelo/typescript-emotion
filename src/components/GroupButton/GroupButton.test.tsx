import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import GroupButton from './GroupButton';

describe('<GroupButton />', () => {
  it('should render with right direction by default', () => {
    const tree = shallow(<GroupButton />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render with left direction', () => {
    const tree = shallow(<GroupButton direction="left" />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
