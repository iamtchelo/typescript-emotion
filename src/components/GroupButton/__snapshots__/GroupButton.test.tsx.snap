// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupButton /> should render with left direction 1`] = `
<Styled(div)
  direction="left"
/>
`;

exports[`<GroupButton /> should render with right direction by default 1`] = `
<Styled(div)
  direction="right"
/>
`;
